### ISSUE TITLE
Thread panic when using Blockhash, but not with DCT

### ISSUE THREAD

**Issue by cryptoquick** [abonander/img_hash#23]
works: 

```rust
let img = image::load_from_memory(&mut buf).unwrap();
let hash = ImageHash::hash(&img, 8, HashType::DCT);
```

doesn't work:

```rust
let img = image::load_from_memory(&mut buf).unwrap();
let hash = ImageHash::hash(&img, 8, HashType::Block);
```

`thread 'main' panicked at 'index out of bounds: the len is 64 but the index is 64', /checkout/src/libcore/slice/mod.rs:2085:14`

versions:
rustc: 1.28-nightly
image = "0.13" # had to peg this to an older version to get it to work

[dependencies.img_hash]
version = "^2.0.1"
features = ["rust-image"]

stack trace:

```
thread 'main' panicked at 'index out of bounds: the len is 64 but the index is 64', /checkout/src/libcore/slice/mod.rs:2085:14
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:227
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:511
   5: std::panicking::continue_panic_fmt
             at libstd/panicking.rs:426
   6: rust_begin_unwind
             at libstd/panicking.rs:337
   7: core::panicking::panic_fmt
             at libcore/panicking.rs:92
   8: core::panicking::panic_bounds_check
             at libcore/panicking.rs:60
   9: <usize as core::slice::SliceIndex<[T]>>::index_mut
             at /checkout/src/libcore/slice/mod.rs:2085
  10: core::slice::<impl core::ops::index::IndexMut<I> for [T]>::index_mut
             at /checkout/src/libcore/slice/mod.rs:1958
  11: <alloc::vec::Vec<T> as core::ops::index::IndexMut<I>>::index_mut
             at /checkout/src/liballoc/vec.rs:1717
  12: img_hash::block::blockhash_slow::{{closure}}
             at /home/~/img_hash-2.0.1/src/block.rs:99
  13: img_hash::rust_image::<impl img_hash::HashImage for image::dynimage::DynamicImage>::foreach_pixel
             at /home/~/img_hash-2.0.1/src/rust_image.rs:89
  14: img_hash::block::blockhash_slow
             at /home/~/img_hash-2.0.1/src/block.rs:72
  15: img_hash::block::blockhash
             at /home/~/img_hash-2.0.1/src/block.rs:29
  16: img_hash::HashType::hash
             at /home/~/img_hash-2.0.1/src/lib.rs:279
  17: img_hash::ImageHash::hash
             at /home/~/img_hash-2.0.1/src/lib.rs:85
  18: app::event
             at src/main.rs:63
  19: nannou::run_loop::process_and_emit_glutin_event
             at /home/~/nannou-0.6.0/src/lib.rs:368
  20: nannou::run_loop
             at /home/~/nannou-0.6.0/src/lib.rs:412
  21: <nannou::Builder<M, E>>::run
             at /home/~/nannou-0.6.0/src/lib.rs:147
  22: app::main
             at src/main.rs:14
  23: std::rt::lang_start::{{closure}}
             at /checkout/src/libstd/rt.rs:74
  24: std::panicking::try::do_call
             at libstd/rt.rs:59
             at libstd/panicking.rs:310
  25: __rust_maybe_catch_panic
             at libpanic_unwind/lib.rs:105
  26: std::rt::lang_start_internal
             at libstd/panicking.rs:289
             at libstd/panic.rs:392
             at libstd/rt.rs:58
  27: std::rt::lang_start
             at /checkout/src/libstd/rt.rs:74
  28: main
  29: __libc_start_main
  30: _start
```

**Comment by abonander** [comment]
Sorry, I've been neglecting this library for a while. I need to bump the version to 3.0.0-alpha.x until the `image` crate releases its 1.0 so upgrades aren't breaking changes. Once I've done that and it's building successfully, I'd like you to see if the panic still occurs. It's probably an off by one error in my code anyway.

**Comment by cryptoquick** [comment]
Understandable. Does `image` have a tracking issue for 1.0?

**Comment by abonander** [comment]
Not that I could find.

**Comment by cryptoquick** [comment]
I left them an issue. Hopefully this will help things along. :+1:

**Comment by abonander** [comment]
Fix published in `2.0.2` and soon `3.0.0-alpha.1` along with some public dependency changes.

**Comment by nbdd0121** [comment]
The OOB access still happens in version 3.1.
